### DISCUSSION TITLE
Another question about D （Tau-U）

### DISCUSSION THREAD

**Discussion by sunflowerecnu** [jazznbass/scan#145] [Q&A] [answered]
[syntax]
try <- scdf(
  c(A = 40, 40, 40, 40, 30, 30, 
    B = 40, 50, 50, 60, 60, 70, 80, 70, 80, 90, 90, 90)
)
tau_u(try, method="complete",tau_method="b") %>% print(complete = TRUE)
[result]
![image](https://github.com/jazznbass/scan/assets/132215966/ea15a668-52db-4d23-9bcc-c86eedbcc3b5)
 
[question]
D 
A vs. B                                   D=70=sqrt(72*(72-4))
Trend A                                 D=10.95=sqrt(15*(15-7))
Trend B                                 OK
A vs. B + TrendB- Trend A    OK
But, 
A vs. B - Trend A            D=108.37(?)
A vs. B + Trend B           D=136.49(?) 

How to compute D(A vs. B - Trend A) and D(A vs. B + Trend B)?

**Comment by jazznbass** [comment, marked as answer]
D is derived from the D statistic of a Kendalls-Tau analysis. For Tau b this is a complicated formula as ties are taken iinto account.
Here is the calculation syntax

```
For A vs. B - Trend A it is: kendall(AB, c(nA:1, rep(nA + 1, nB)), tau_method = tau_method)
For A vs. B + Trend B it is: kendall(AB, c(rep(0, nA), (nA + 1):nAB), tau_method=tau_method)
```

See here for more details:
Parker, R. I., Vannest, K. J., & Davis, J. L. (2011). Effect Size in Single-Case Research: A Review of Nine Nonoverlap Techniques. Behavior Modification, 35(4), 303–322. https://doi.org/10.1177/0145445511399147

**Reply by sunflowerecnu** [reply]
Thanks a lot.
